Macro redefinition warnings from `limits.h`

```sh-session
$ zig build test
Test [70/216] limits header... /home/vexu/Documents/zig/sfcc/include/limits.h:21:9: warning: 'SCHAR_MIN' macro redefined [-Wmacro-redefined]
#define SCHAR_MIN (-__SCHAR_MAX__-1)
        ^
/home/vexu/Documents/zig/sfcc/include/limits.h:22:9: warning: 'SHRT_MIN' macro redefined [-Wmacro-redefined]
#define SHRT_MIN  (-__SHRT_MAX__ -1)
        ^
/home/vexu/Documents/zig/sfcc/include/limits.h:23:9: warning: 'INT_MIN' macro redefined [-Wmacro-redefined]
#define INT_MIN   (-__INT_MAX__  -1)
        ^
/home/vexu/Documents/zig/sfcc/include/limits.h:24:9: warning: 'LONG_MIN' macro redefined [-Wmacro-redefined]
#define LONG_MIN  (-__LONG_MAX__ -1L)
        ^
/home/vexu/Documents/zig/sfcc/include/limits.h:26:9: warning: 'UCHAR_MAX' macro redefined [-Wmacro-redefined]
#define UCHAR_MAX (__SCHAR_MAX__*2  +1)
        ^
/home/vexu/Documents/zig/sfcc/include/limits.h:27:9: warning: 'USHRT_MAX' macro redefined [-Wmacro-redefined]
#define USHRT_MAX (__SHRT_MAX__ *2  +1)
        ^
/home/vexu/Documents/zig/sfcc/include/limits.h:28:9: warning: 'UINT_MAX' macro redefined [-Wmacro-redefined]
#define UINT_MAX  (__INT_MAX__  *2U +1U)
        ^
/home/vexu/Documents/zig/sfcc/include/limits.h:29:9: warning: 'ULONG_MAX' macro redefined [-Wmacro-redefined]
#define ULONG_MAX (__LONG_MAX__ *2UL+1UL)
        ^
/home/vexu/Documents/zig/sfcc/include/limits.h:35:9: warning: 'CHAR_BIT' macro redefined [-Wmacro-redefined]
#define CHAR_BIT  __CHAR_BIT__
        ^
/home/vexu/Documents/zig/sfcc/include/limits.h:85:9: warning: 'CHAR_MAX' macro redefined [-Wmacro-redefined]
#define CHAR_MAX __SCHAR_MAX__
        ^
10 warnings generated.
$ yay -Q glibc
glibc 2.37-2

```

--- a/include/limits.h
+++ b/include/limits.h
@@ -18,20 +18,35 @@
 #undef LONG_MAX
 #define LONG_MAX  __LONG_MAX__
 
+#undef SCHAR_MIN
 #define SCHAR_MIN (-__SCHAR_MAX__-1)
+
+#undef SHRT_MIN
 #define SHRT_MIN  (-__SHRT_MAX__ -1)
+
+#undef INT_MIN
 #define INT_MIN   (-__INT_MAX__  -1)
+
+#undef LONG_MIN
 #define LONG_MIN  (-__LONG_MAX__ -1L)
 
+#undef UCHAR_MAX
 #define UCHAR_MAX (__SCHAR_MAX__*2  +1)
+
+#undef USHRT_MAX
 #define USHRT_MAX (__SHRT_MAX__ *2  +1)
+
+#undef UINT_MAX
 #define UINT_MAX  (__INT_MAX__  *2U +1U)
+
+#undef ULONG_MAX
 #define ULONG_MAX (__LONG_MAX__ *2UL+1UL)
 
 #ifndef MB_LEN_MAX
 #define MB_LEN_MAX 1
 #endif
 
+#undef CHAR_BIT
 #define CHAR_BIT  __CHAR_BIT__
 
 #if defined(__STDC_VERSION__) && __STDC_VERSION__ >= 202311L
@@ -77,6 +92,8 @@
 
 #endif  /*  defined(__STDC_VERSION__) && __STDC_VERSION__ >= 202311L */
 
+#undef CHAR_MIN
+#undef CHAR_MAX
 #ifdef __CHAR_UNSIGNED__
 #define CHAR_MIN 0
 #define CHAR_MAX UCHAR_MAX

